pkg/deviceplugins/usb_device_plugin.go

```diff
@@ -0,0 +1,469 @@
+package deviceplugins
+
+/* This file was part of the KubeVirt project, copied to this project
+ * to get around private package issues.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2024 SUSE, LLC.
+ *
+ */
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"k8s.io/apimachinery/pkg/util/rand"
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/log"
+	"kubevirt.io/kubevirt/pkg/safepath"
+	"kubevirt.io/kubevirt/pkg/util"
+	pluginapi "kubevirt.io/kubevirt/pkg/virt-handler/device-manager/deviceplugin/v1beta1"
+)
+
+var (
+	pathToUSBDevices = "/sys/bus/usb/devices"
+)
+
+var discoverLocalUSBDevicesFunc = discoverPluggedUSBDevices
```

Comment:
this rename is confusing, and makes it harder to follow? is there a specific reason for this?

Reply:
Oh, it's for testing when doing mock in original kubevirt project. But, i didn't copy the test file here.